server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/NodeInfo.java

```diff
@@ -227,7 +235,16 @@ private <T extends ReportingService.Info> void addInfoIfNonNull(Class<T> clazz,
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
-        Version.writeVersion(version, out);
+        if (out.getTransportVersion().onOrAfter(TransportVersions.NODE_INFO_VERSION_AS_STRING)) {
+            out.writeString(version);
+        } else {
+            try {
+                Version.writeVersion(Version.fromString(version), out);
+            } catch (IllegalArgumentException e) {
```

Comment:
If we can't parse the version, something is wrong. Let's not swallow the exception? But why would we ever expect this since we've successfully created this NodeInfo already? Seems like something we could validate earlier rather than every time we write out the info to a stream.

Reply:
I would not expect this: I would not expect a mix of nodes with semantic/non-semantic version.
But there may be one little window I think, in serverless: we make this change in 8.12.0, then we also introduce a change where we start using non-semantic versions (hash). We run a mixed cluster (8.11 - 8.12) during an upgrade -  could we end up reading a hash and trying to convert it to a version?
Initially I did not try to catch anything. But I saw @williamrandolph PR and thought it was a good idea to be consistent. But I'm fine either way.

Reply:
I don't think that scenario can happen in serverless. For now, the version string we use in serverless is still the on-prem version, we just don't want to expose/rely on it anywhere. So this still should be a semantic version there, for now at least.

Reply:
Good, and it's on a version-gated path so we should be good for the future too. I will remove the catch.